### DISCUSSION TITLE
EthereumProvider.init

### DISCUSSION THREAD

**Discussion by Amazing123511** [WalletConnect/walletconnect-monorepo#2680] [Developer Support]
let provider = await EthereumProvider.init({
      projectId: projectId, // REQUIRED your projectId
      chains: ["5"], // REQUIRED chain ids
      optionalChains: ["1"],
      showQrModal: true, // REQUIRED set to "true" to use @web3modal/standalone,
      methods: ["eth_accounts", "eth_requestAccounts", "eth_sign", "personal_sign", "eth_sendTransaction", "eth_chainId", "wallet_switchEthereumChain", "sendCustomRequest", "wallet_addEthereumChain"], // OPTIONAL ethereum methods
      events: [
        "chainChanged",
        "accountsChanged",
        "connect",
        "session_event",
        "display_uri",
        "disconnect",
      ], // OPTIONAL ethereum events
    },

After I initialize this method, I want to switch the chain later, but it is not supported

**Comment by glitch-txs** [comment]
Hi, which chain are you trying to change to and which wallet are you using?

**Reply by Amazing123511** [reply]
I use the metaMask wallet. When I was testing, I wanted him to pull up the wallet to switch the Ethereum main chain. Use this wallet_switchEthereumChain method, but it prompts, this chain is not supported? I can pull it up when I use 1.0

**Reply by Amazing123511** [reply]
Can't I write multiple chain ids in chains when I initialize?

**Reply by andriirak** [reply]
same for me

**Reply by Amazing123511** [reply]
Have you solved it?

**Reply by andriirak** [reply]
I didn`t fix, created a new bug for MM mobile https://github.com/MetaMask/metamask-mobile/issues/6655

**Reply by finessevanes** [reply]
@Amazing123511 are you using MetaMask browser or mobile to connect?

**Reply by Amazing123511** [reply]
mobile

**Comment by windatang** [comment]
Dev for web-app & debug with mobile-app  I report that 

Register multiple Chain will reject by app
[issues/2491](https://github.com/WalletConnect/walletconnect-monorepo/issues/2491)
Register one Chain only create instance is working otherwise it no way to switch (@web3modal we3-modal) 
[issues/2661](https://github.com/WalletConnect/walletconnect-monorepo/issues/2661)
- try the way reset and setChainIds 
- try the way  use the `"@walletconnect/modal"` -> `@web3modal/core` ->  `ConfigCtrl.setConfig`
Both are not working
